ouroboros-network/src/Ouroboros/Network/PeerSelection.hs

```diff
@@ -0,0 +1,42 @@
+{-# LANGUAGE ExplicitNamespaces #-}
+
+module Ouroboros.Network.PeerSelection
+  ( module Governor
+  , module PeerSelection
+  , module PeerSelection.Types
+  , module PeerSelection.PublicRootPeers
+  , module PeerSelection.PeerStateActions
+  , module PeerSelection.PeerSelectionActions
+  , module PeerSelection.RelayAccessPoint
+  , module PeerSelection.LedgerPeers
+  , module PeerSelection.PeerMetrics
+  , module PeerSelection.Churn
+  , module PeerSelection.PeerAdvertise
+  , module PeerSelection.PeerSharing
+  ) where
+
+import Ouroboros.Network.PeerSelection.Churn as PeerSelection.Churn
+-- Only essential `Governor` types.
+import Ouroboros.Network.PeerSelection.Governor as Governor
+           (DebugPeerSelection (..), PeerSelectionActions,
+           PeerSelectionInterfaces (..), PeerSelectionPolicy (..),
+           PeerSelectionState, PeerSelectionTargets (..), PeerStateActions,
+           PickPolicy)
+import Ouroboros.Network.PeerSelection.LedgerPeers as PeerSelection.LedgerPeers
+           (AfterSlot (..), IsBigLedgerPeer (..), LedgerPeerSnapshot (..),
+           LedgerPeers (..), LedgerPeersConsensusInterface (..),
+           LedgerPeersKind (..), NumberOfPeers (..), TraceLedgerPeers (..),
+           UseLedgerPeers (..), WithLedgerPeersArgs (..), withLedgerPeers)
+import Ouroboros.Network.PeerSelection.PeerAdvertise as PeerSelection.PeerAdvertise
+import Ouroboros.Network.PeerSelection.PeerMetric as PeerSelection.PeerMetrics
+           (PeerMetrics, PeerMetricsConfiguration (..), ReportPeerMetrics (..),
+           newPeerMetric, newPeerMetric', nullMetric, reportMetric)
+import Ouroboros.Network.PeerSelection.PeerSelectionActions as PeerSelection.PeerSelectionActions
+import Ouroboros.Network.PeerSelection.PeerSharing as PeerSelection.PeerSharing
+import Ouroboros.Network.PeerSelection.PeerStateActions as PeerSelection.PeerStateActions
+import Ouroboros.Network.PeerSelection.PublicRootPeers as PeerSelection.PublicRootPeers
+           (PublicRootPeers)
+import Ouroboros.Network.PeerSelection.RelayAccessPoint as PeerSelection.RelayAccessPoint
+           (DomainAccessPoint (..), IP (..), PortNumber, RelayAccessPoint (..))
+import Ouroboros.Network.PeerSelection.Types as PeerSelection
+import Ouroboros.Network.PeerSelection.Types as PeerSelection.Types
```

Comment:
I don't like this... I prefer for the imports to come from the place where the expression is defined. Meta modules like these are confusing for large codebases

Reply:
Another solution could be to have explicit imports in the meta modules, but I am not sure if it's worth it in times of `hls` / `ghc-tags-plugin` etc.

Reply:
Right but, I also use HLS to manage automated imports for me and I usually import from the modules which defined the expression/function. So my worry is that we'll have inconsistencies still in the imports. At least I think the name should be more obvious so I know I should import from the `Meta`/`General`/`Imports`/`All` modules